File: Behavioral/Strategy.py
class SortStrategy:
    def sort(self, dataset):
        pass

class BubbleSortStrategy(SortStrategy):
    def sort(self, dataset):
        print 'Sorting using bubble sort'

        return dataset

class QuickSortStrategy(SortStrategy):
    def sort(self, dataset):
        print 'Sorting using quick sort'
        return dataset

class Sorter:
    _sorter = None

    def __init__(self, sorter):
        self._sorter = sorter

    def sort(self, dataset):
        return self._sorter.sort(dataset)

dataset = [1, 5, 4, 3, 2, 8]

sorter = Sorter(BubbleSortStrategy())
sorter.sort(dataset)

sorter = Sorter(QuickSortStrategy())
sorter.sort(dataset)

File: examples-per-language/php/behavioral/Strategy.php
<?php

use designPatternsForHumans\behavioral\Strategy\BubbleSortStrategy;
use designPatternsForHumans\behavioral\Strategy\QuickSortStrategy;
use designPatternsForHumans\behavioral\Strategy\Sorter;

require_once __DIR__ . '/autoload.php';

$dataset = [1, 5, 2, 6, 3];

$sorter = new Sorter(new BubbleSortStrategy());
$sorter->sort($dataset);

$sorter = new Sorter(new QuickSortStrategy());
$sorter->sort($dataset);
